### PR TITLE
Add dynamics interface

### PR DESCRIPTION
**Pull request by lewispb**
Add and configure the dynamics crm gem.

To create or update a model we'd do something like this:

```
account = Account.find(2)
client = CrmInterface.new(account).sync
```

### CODE REVIEW

File: lib/crm_interface.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  def update
+    client.update(resource_class.downcase, resouce.dynamics_id, resource_hash)
+  end
+
+  def create
+    client.create(resource_class.downcase, resource_hash)
+  end
+
+  def client
```
Comment (autotelik) [inline lib/crm_interface.rb]: maybe should be private ?

File: lib/crm_interface.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+class CrmInterface
+
+  attr_reader :resource, :resource_class, :resource_hash
+
+  def initialize(resource = nil)
```
Comment (autotelik) [inline lib/crm_interface.rb]: ever valid to create an interface without a resource ? .. cos looks like you could and then call sync and get null object error

**Comment by autotelik** [comment]
couple of minor comments to take or leave - :ship:

File: lib/crm_interface.rb (hunk 2 above)
Comment (lewispb) [inline lib/crm_interface.rb]: Agreed

**Pull request merged**